### ISSUE TITLE
HAML SASS Directive fails when passed an 8-digit hex value

### ISSUE THREAD

**Issue by dlinch** [rtomayko/tilt#337]
tl;dr: The HAML Sass engine fails to parse 8-digit hex codes as valid css.

I originally opened this issue in the HAML repo, but it was closed and I was told HAML delegates the Sass handling to Tilt. [https://github.com/haml/haml/issues/997#issuecomment-442064901](https://github.com/haml/haml/issues/997#issuecomment-442064901)

We are working on a project that allows theming, and we're injecting a base color into the application head tag to build our theme out from.

Right now it looks like:

```
:sass
  $primary-color: #{@color}
  $secondary-color: #{@second_color}
  # Do stuff with the colors
```

Our primary color comes through as a standard hex value `#000000` or `#000`, but our secondary is just our primary color that we programmatically add an alpha value onto. (So it will look like `#00000000` or `#0000`). When we go to load the page, the HAML engine errors and the error message points to our 8-digit hex value as invalid CSS.
4-digit and 8-digit hex codes have pretty wide adoption at the moment: [https://caniuse.com/#search=hex](https://caniuse.com/#search=hex)
So I would argue that Tilt should support it.

I found a closed issue from 8 years ago in the haml repo that might point a contributor in the right direction [https://github.com/haml/haml/issues/168](https://github.com/haml/haml/issues/168)

**Comment by judofyr** [comment]
The latest version of Tilt (2.0.9) supports the latest SassC engine:

```ruby
# latest-sass.rb
require 'bundler/inline'

gemfile do
  source 'https://rubygems.org'
  gem 'tilt', '2.0.9'
  gem 'sassc', '2.0.0'  # <- This is important! You should use `sassc` and not `sass` now
end

t = Tilt.new('hello.sass') { <<~'EOF' }
$primary-color: #FF00FFF5;

body
  background: $primary-color;
EOF

puts t.render
```

And here you see that the latest Sass version currently parses it as string, but some time in the future will support parsing it as a color:

```
$ ruby latest-sass.rb
DEPRECATION WARNING on line 1, column 26 of hello.sass:
The value "#FF00FFF5" is currently parsed as a string, but it will be parsed as a color in
future versions of Sass. Use "unquote('#FF00FFF5')" to continue parsing it as a string.

body {
  background: #FF00FFF5; }
```

Not quite sure what "future version of Sass" means here though. But look out for new releases to [sassc](https://rubygems.org/gems/sassc)

**Comment by dlinch** [comment]
Ok, I will close this and open an issue with SassC. Sorry I didn't see this for so long.
